File: src/main/java/com/myapp/dto/UserStats.java
package com.myapp.dto;

import com.querydsl.core.annotations.QueryProjection;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class UserStats {

    private final long micropostCnt;
    private final long followingCnt;
    private final long followerCnt;

    @QueryProjection
    public UserStats(long micropostCnt, long followingCnt, long followerCnt) {
        this.micropostCnt = micropostCnt;
        this.followingCnt = followingCnt;
        this.followerCnt = followerCnt;
    }

}

File: src/main/kotlin/com/myapp/domain/UserStats.kt
package com.myapp.domain

import com.myapp.generated.tables.records.UserStatsRecord

data class UserStats(
    val micropostCnt: Long,
    val followingCnt: Long,
    val followerCnt: Long
) {
    constructor(record: UserStatsRecord) : this(
        micropostCnt = record.micropostCnt,
        followerCnt = record.followerCnt,
        followingCnt = record.followingCnt
    )
}